File: src/main/java/Swap.java
public class Swap {
  public static void main(String[] args) {
    int a = -1;
    int y = 5;

    int originalA = a;
    int originalY = y;

    // swap

    a = a + y;
    y = a - y;
    a = a - y;

    System.out.println(a == originalY);
    System.out.println(y == originalA);
  }
}

File: src/main/java/Swap.kt

    fun main() {
        var a = -1
        var y = 5

        val originalA = a
        val originalY = y

        // swap

        a = a + y
        y = a - y
        a = a - y

        println(a == originalY)
        println(y == originalA)
    }

